vowpalwabbit/example.cc

```diff
@@ -4,8 +4,39 @@
 #include <cstdint>
 #include <algorithm>
 
+#include "example.h"
 #include "gd.h"
 
+example::example()
+{
+  memset(&l, 0, sizeof(polylabel));
+  memset(&pred, 0, sizeof(polyprediction));
+  tag = v_init<char>();
+}
+
+example::~example()
```

Comment:
You'll need to add an r-value constructor too, otherwise you can end up getting double frees and the like